src/modules/Web3mail/context/web3mail.tsx

```diff
@@ -0,0 +1,250 @@
+  const { account } = useContext(TalentLayerContext);
+  const [platformHasAccess, setPlatformHasAccess] = useState(false);
+  const [emailIsProtected, setEmailIsProtected] = useState(false);
+  const [dataProtector, setDataProtector] = useState<IExecDataProtector>();
+  const [web3mail, setWeb3mail] = useState<IExecWeb3mail>();
+  const [isFetching, setIsFetching] = useState<boolean>(false);
+  const [protectedEmail, setProtectedEmail] = useState<ProtectedData | undefined>();
+  const [emailGrantedAccess, setEmailGrantedAccess] = useState<GrantedAccess | undefined>();
+
+  log('Web3MailProvider ---- call', {
+    dataProtector,
+    account,
+    protectedEmail,
+    emailGrantedAccess,
+    emailIsProtected,
+    platformHasAccess,
+  });
+
+  /*
+   * @what: Init The dataProtector and web3mail modules
+   * @when: Execute it only once after user is connected, and only if user switch wallet
+   */
+  useEffect(() => {
+    const fetchData = async () => {
+      if (process.env.NEXT_PUBLIC_ACTIVE_WEB3MAIL == 'false') {
+        return;
+      }
+      if (!account?.isConnected || dataProtector || web3mail) {
+        return;
+      }
+
+      log('Web3MailProvider ---- Init The dataProtector', { dataProtector, account });
+
+      // Note: RPC error if using provider
+      // setDataProtector(new IExecDataProtector(provider as providers.ExternalProvider));
+      // setWeb3mail(new IExecWeb3mail(provider as providers.ExternalProvider));
+
+      // TODO: cause error on runtime: Error: Unsupported ethProvider
+      const provider2 = await account.connector?.getProvider();
+      setDataProtector(new IExecDataProtector(provider2));
+      setWeb3mail(new IExecWeb3mail(provider2));
+
+      // const provider2 = await account.connector?.getProvider();
+      // setDataProtector(new IExecDataProtector(publicClient));
+      // setWeb3mail(new IExecWeb3mail(publicClient));
+
+      // This work but better in our case to use account from the hook
+      // const provider2 = await getAccount().connector?.getProvider();
+      // setDataProtector(new IExecDataProtector(provider2));
+      // setWeb3mail(provider2);
+    };
+    fetchData();
+  }, [account]);
+
+  /*
+   * @what: Check if platform has access to the user email
+   * @when: Execute it only once after user is connected, and only if user switch wallet
+   */
+  useEffect(() => {
+    if (!dataProtector || !web3mail || !account?.isConnected || isFetching) return;
+
+    log('Web3MailProvider ---- check if platform has access ?', {
+      dataProtector,
+      web3mail,
+    });
+
+    setIsFetching(true);
+
+    const fetchData = async () => {
+      log('Web3MailProvider ----  before fetching');
+      const protectedData = await dataProtector.fetchProtectedData({
+        owner: account.address,
+        requiredSchema: {
+          email: 'string',
+        },
+      });
+
+      const protectedEmail = protectedData.find(item => item.name === 'Untitled');
```

Comment:
Why keep "Untitled ?" Shouldn"t we set a TL Standard for this ?

Reply:
it's just to be more easy to the user, one wallet, one default shared email